server/encryption/localkey_handler_test.go

```diff
@@ -0,0 +1,194 @@
+package encryption
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Ensure that a data key can be generated
+func TestGenerateDataKeyCorrectly(t *testing.T) {
+	// Set a random AES key as master key
+	os.Setenv("LOCAL_MASTER_KEY", "+KbPeShVmYq3t6w9")
+
+	// Given a key handler
+	keyHandler, err := NewLocalEncriptionHandler()
+	require.NoError(t, err)
+
+	// Generate DKS
+	dks, err := keyHandler.generateDKS()
+
+	// Expect DKS is generated without error
+	require.NoError(t, err)
+	require.NotEmpty(t, dks)
+}
+
+// ENsure that key can be wrapped by using master key
+func TestWrapDataKeyCorrectly(t *testing.T) {
+	// Set a random AES key as master key
+	os.Setenv("LOCAL_MASTER_KEY", "t7w!z%C*F-JaNcRf")
+
+	// Given a key handler
+	keyHandler, err := NewLocalEncriptionHandler()
+	require.NoError(t, err)
+
+	// Generate DKS
+	dks, err := keyHandler.generateDKS()
+
+	// Expect DKS is generated without error
+	require.NoError(t, err)
+	require.Equal(t, DataKeyLength, len(dks))
+
+	// Start wrapping DKS
+	wrappedDKS, err := keyHandler.wrapDKS(dks)
+
+	// Expect DKS is wrapped without error
+	require.NoError(t, err)
+	require.NotEmpty(t, wrappedDKS)
+	// Expect wrapped DKS should not be the same
+	// as plain text DKS
+	require.NotEqual(t, dks, wrappedDKS)
+}
+
+// Ensure that key can be unwrapped by using master key
+func TestUnWrapDataKeyCorrectly(t *testing.T) {
+	// Set a random AES key as master key
+	os.Setenv("LOCAL_MASTER_KEY", "t7w!z%C*F-JaNcRf")
+
+	// Given a key handler
+	keyHandler, err := NewLocalEncriptionHandler()
+	require.NoError(t, err)
+
+	// Generate DKS
+	dks, err := keyHandler.generateDKS()
+
+	// Expect DKS is generated without error
+	require.NoError(t, err)
+	require.Equal(t, DataKeyLength, len(dks))
+
+	// Start wrapping DKS
+	wrappedDKS, err := keyHandler.wrapDKS(dks)
+	require.NoError(t, err)
+
+	// Unwrap
+	key, err := keyHandler.unwrapDKS(wrappedDKS)
+	require.NoError(t, err)
+
+	// Expect the unwrapped key is actually correct
+	require.Equal(t, dks, key)
+}
+
+// Ensure encryptions can encrypt the message
+func TestEncryption(t *testing.T) {
+	// Set a random AES key as master key
+	os.Setenv("LOCAL_MASTER_KEY", "t7w!z%C*F-JaNcRf")
+
+	// Given a key handler
+	keyHandler, err := NewLocalEncriptionHandler()
+	require.NoError(t, err)
+
+	// Given sample data
+	plaintext := []byte("exampleplaintext")
+
+	// Given a sample data key
+
+	dks, err := keyHandler.generateDKS()
+	require.NoError(t, err)
+
+	ciphertext, err := keyHandler.encryptData(dks, plaintext)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, ciphertext)
+
+	// Expect cipher  text should not be the same
+	// as plain text
```

Comment:
```suggestion
	// expect cipher text should not be the same as plain text
```

(I'd lowercase the comments in this file)